src/thread.rs

```diff
@@ -1,111 +1,37 @@
-//! Scoped thread.
+//! Threads that can borrow variables from the stack.
 //!
-//! # Examples
+//! Create a scope when spawned threads need to access variables on the stack:
 //!
-//! A basic scoped thread:
-//!
-//! ```
-//! crossbeam_utils::thread::scope(|scope| {
-//!     scope.spawn(|| {
-//!         println!("Hello from a scoped thread!");
-//!     });
-//! }).unwrap();
-//! ```
-//!
-//! When writing concurrent Rust programs, you'll sometimes see a pattern like this, using
-//! [`std::thread::spawn`]:
-//!
-//! ```ignore
-//! let array = [1, 2, 3];
-//! let mut guards = vec![];
-//!
-//! for i in &array {
-//!     let guard = std::thread::spawn(move || {
-//!         println!("element: {}", i);
-//!     });
-//!
-//!     guards.push(guard);
-//! }
-//!
-//! for guard in guards {
-//!     guard.join().unwrap();
-//! }
-//! ```
-//!
-//! The basic pattern is:
-//!
-//! 1. Iterate over some collection.
-//! 2. Spin up a thread to operate on each part of the collection.
-//! 3. Join all the threads.
-//!
-//! However, this code actually gives an error:
-//!
-//! ```text
-//! error: `array` does not live long enough
-//! for i in &array {
```

Comment:
We're removing the (hypothetical) error message here.  I'm not strongly against this change, but I think this error message can help readers to understand the motivation behind scoped threads.  What do you think?

Reply:
I've just added a section that illustrates why we need scoped threads. Is it better now?